modules/lib/src/main/resources/assets/js/app/ui2/dialog/ConfirmationDialog.tsx

```diff
@@ -0,0 +1,133 @@
+import {Button, Dialog, cn} from '@enonic/ui';
+import type {ComponentPropsWithoutRef, ReactElement, ReactNode} from 'react';
+import {createContext, useCallback, useContext, useMemo, useState} from 'react';
+import {i18n} from '@enonic/lib-admin-ui/util/Messages';
+import {forwardRef} from 'react';
+
+//
+// Confirmation dialog context
+//
+
+type ConfirmationDialogContextValue = {
+    unlocked: boolean;
+    setUnlocked: (unlocked: boolean) => void;
+};
+
+const ConfirmationDialogContext = createContext<ConfirmationDialogContextValue | null>(null);
+
+const useConfirmationDialog = (): ConfirmationDialogContextValue => {
+    const ctx = useContext(ConfirmationDialogContext);
+    if (!ctx) {
+        // Safe default: unlocked to keep buttons enabled if used outside Content wrapper
+        return {unlocked: true, setUnlocked: () => void 0};
+    }
+    return ctx;
+};
+
+type ConfirmationDialogProviderProps = {
+    defaultUnlocked?: boolean;
+    children?: ReactNode;
+};
+
+const ConfirmationDialogProvider = ({defaultUnlocked = true, children}: ConfirmationDialogProviderProps): ReactElement => {
+    const [unlocked, setUnlocked] = useState<boolean>(defaultUnlocked);
+
+    const value = useMemo<ConfirmationDialogContextValue>(() => ({unlocked, setUnlocked}), [unlocked]);
+
+    return <ConfirmationDialogContext.Provider value={value}>{children}</ConfirmationDialogContext.Provider>;
+};
+
+//
+// Content: wraps Dialog.Content with provider to keep shared state
+//
+
+export type ConfirmationDialogContentProps = {
+    defaultUnlocked?: boolean;
+    className?: string;
+    children?: ReactNode;
+} & ComponentPropsWithoutRef<typeof Dialog.Content>;
+
+const ConfirmationDialogContent = forwardRef<HTMLDivElement, ConfirmationDialogContentProps>(
+    ({defaultUnlocked = true, className, children, ...props}, ref): ReactElement => {
+        return (
+            <Dialog.Content ref={ref}
+                            className={cn('!outline-none [&_button]:!outline-none gap-3 min-w-170 max-w-fit', className)} {...props}>
```

Comment:
Still keeping fixes for outline de to conflicts with legacy styles? Just ignore it for now.